components/script/dom/domrectlist.rs

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use dom_struct::dom_struct;
+
+use crate::dom::bindings::cell::DomRefCell;
+use crate::dom::bindings::codegen::Bindings::DOMRectListBinding::DOMRectListMethods;
+use crate::dom::bindings::reflector::{reflect_dom_object_with_proto, DomObject, Reflector};
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::domrect::DOMRect;
+use crate::dom::window::Window;
+use crate::script_runtime::CanGc;
+
+#[dom_struct]
+pub struct DOMRectList {
+    reflector_: Reflector,
+    rects: DomRefCell<Vec<DomRoot<DOMRect>>>,
```

Comment:
Let's store `Vec<Dom<DOMRect>>` instead. That makes a bit less work for the garbage collector.